lib/pbench/server/database/models/api_key.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional
+
+from flask import current_app
+import jwt
+from sqlalchemy import Column, ForeignKey, String
+from sqlalchemy.orm import relationship
+
+from pbench.server.database.database import Database
+from pbench.server.database.models import TZDateTime
+from pbench.server.database.models.users import User
+
+
+class APIKeyError(Exception):
+    """A base class for errors reported by the APIKey class."""
+
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return repr(self.message)
+
+
+class APIKeys(Database.Base):
+    """Model for storing the API key associated with a user."""
+
+    __tablename__ = "api_keys"
+    api_key = Column(
+        String(500), primary_key=True, unique=True, nullable=False, index=True
+    )
+    created = Column(TZDateTime, nullable=False, default=TZDateTime.current_time)
+    # ID of the owning user
+    user_id = Column(String, ForeignKey("users.id"), nullable=False)
+
+    # Indirect reference to the owning User record
+    user = relationship("User")
+
+    def add(self):
+        """Add an api_key object to the database."""
+        try:
+            Database.db_session.add(self)
+            Database.db_session.commit()
+        except Exception as e:
+            Database.db_session.rollback()
+            self.logger.error("Can't add {} to DB: {}", str(self), str(e))
```

Comment:
Interestingly, there's no `__str__` method defined, so the `str(self)` is not going to generate anything useful.